### ISSUE TITLE
Primal Decomposition non-convergence with ExaGO

### ISSUE THREAD

**Issue by tabholt** [LLNL/hiop#615]
When using HiOp pri-dec solver with Ipopt as the sub-problem solver in ExaGO scopflow I am running into non-convergence issues on the HiOp side. This problem seems to occur when the grid is tight, ie the unit commitment has few enough generators such that the system is close to needing to shed load. 

However I can confirm that load-shedding is not necessary in these contingencies when solving them as a single system with Ipopt. I can also confirm that if we activate all of the generators in the provided case file then it will converge. 

I am using Ipopt 3.13.5 and HiOp 0.7.1 as available on the ORNL Crusher machine.

Here is how the problem can be re-produced:

```
contfile="ACTIVSg500_contingencies.cont"
datafile="datafiles/ACTIVSg500_modified.m"
n_conts=3
n_ranks=$((n_conts+1))

srun -n $n_ranks scopflow -netfile $datafile -scopflow_enable_multicontingency 1 -opflow_include_loadloss_variables 1 -ctgcfile $contfile -scopflow_Nc $n_conts -print_output 1 -scopflow_solver HIOP -scopflow_subproblem_solver IPOPT -save_output
```

Both the datafile and the contingencies file are attached. (Changed to .txt to upload to github.)
[ACTIVSg500_modified.m.txt](https://github.com/LLNL/hiop/files/11235520/ACTIVSg500_modified.m.txt)
[ACTIVSg500_contingencies.cont.txt](https://github.com/LLNL/hiop/files/11235523/ACTIVSg500_contingencies.cont.txt)

**Comment by tabholt** [comment]
@jwang125 Slaven Peles suggested I tag you :)

**Comment by cnpetra** [comment]
just to confirm, are you seeing the non-convergence issues in HiOp-PriDec, correct? Not in HiOp as the subproblem solver. Sending us the log(s) (emailing them is fine) would help us greatly.

**Comment by tabholt** [comment]
Yes this is correct. HiOp Pridec with Ipopt as the sub-problem solver. The logs are infinitely long with all of the runs of Ipopt in there. Are there certain verbosity levels you would like so that I can give you a useful log? Could let me know how to set those?

**Comment by tabholt** [comment]
I have emailed a log with 55 iterations with vebosity_level 3 on HiOp and ipopt supressed. I'll also add it here. 

If you would like me to run with other settings please let me know. 

[hiop_non_convergence.log](https://github.com/LLNL/hiop/files/11236470/hiop_non_convergence.log)

**Comment by jwang125** [comment]
Hi Tim, based on your log, I suggest you do `set_alpha_min(1e3)` for the hiopAlgPrimalDecomp class object and see if the results improve. I need to add in general however, neither primal decomp nor exago is implemented in a way that guarantees convergence. So there could be issues with difficult cases. Thank you!

**Comment by cnpetra** [comment]
closing this, feel free to reopen if you need additional assistance

**Comment by huojiangming** [comment]
Cross-referencing one mechanism that would fit this, still in `develop` (fc0e881): #794 — PriDec solves the master problem once per decomposition iteration on the same `hiopNlpSparse`, and `apply_scaling` is applied again on every `run()`, so the gradient scaling compounds across iterations. Creating a fresh NLP object per solve avoids it. We see the master run to its iteration limit on a small case (case9, 4 contingencies, GPU sub-problems) rather than converge.
